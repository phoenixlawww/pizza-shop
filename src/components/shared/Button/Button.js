import PropTypes from 'prop-types'
import classnames from 'classnames'

import './Button.scss'

export const BUTTON_VARIENTS = {
  PRIMARY: 'PRIMARY',
  SECONDARY: 'SECONDARY',
  LINK: 'LINK'
}

function Button (props) {
  const { className, label, varient, disabled, ...otherProps } = props

  return (
    <button
      className={classnames('s-button', className, {
        's-button-primary': varient === BUTTON_VARIENTS.PRIMARY,
        's-button-secondary': varient === BUTTON_VARIENTS.SECONDARY,
        's-button-link': varient === BUTTON_VARIENTS.LINK
      })}
      disabled={disabled}
      {...otherProps}
    >
      {label}
    </button>
  )
}

Button.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  varient: PropTypes.oneOf(['PRIMARY', 'SECONDARY', 'LINK']),
  disabled: PropTypes.bool
}
Button.defaultProps = {
  varient: 'PRIMARY',
  disabled: false
}

export default Button
