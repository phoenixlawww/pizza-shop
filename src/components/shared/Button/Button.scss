@import "~styles/tools";

.s-button {
  display: flex;
  align-items: center;
  justify-content: center;
  max-width: 100%;
  min-height: 40px;
  padding: 5px 30px;
  font-size: 14px;
  line-height: 1.2;
  text-align: center;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 5px;

  &:hover {
    text-decoration: underline;
    cursor: pointer;
  }

  &.s-button-primary {
    color: $white;
    background-color: $dark-grey;
  }
  &.s-button-secondary {
    color: $black;
    border: 1px solid $dark-grey;
  }
  &.s-button-link {
    color: $black;
  }

  &[disabled] {
    cursor: not-allowed;
    opacity: .5;

    &:hover {
      text-decoration: none;
    }
  }
}
