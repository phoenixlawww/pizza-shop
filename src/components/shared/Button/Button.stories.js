import Button, { BUTTON_VARIENTS } from 'components/shared/Button'

export default {
  component: Button,
  title: 'Shared/Button',
  args: {
    label: 'Button',
    onClick: () => alert('clicked')
  }
}

const Template = (args) => <Button {...args} />

export const Primary = Template.bind({})
Primary.args = {
  varient: BUTTON_VARIENTS.PRIMARY
}
export const PrimaryDisabled = Template.bind({})
PrimaryDisabled.args = {
  varient: BUTTON_VARIENTS.PRIMARY,
  disabled: true
}

export const Secondary = Template.bind({})
Secondary.args = {
  varient: BUTTON_VARIENTS.SECONDARY
}
export const SecondaryDisabled = Template.bind({})
SecondaryDisabled.args = {
  varient: BUTTON_VARIENTS.SECONDARY,
  disabled: true
}

export const Link = Template.bind({})
Link.args = {
  varient: BUTTON_VARIENTS.LINK
}
export const LinkDisabled = Template.bind({})
LinkDisabled.args = {
  varient: BUTTON_VARIENTS.LINK,
  disabled: true
}
