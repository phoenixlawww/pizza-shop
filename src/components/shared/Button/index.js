import Button, { BUTTON_VARIENTS } from './Button'
export default Button
export { BUTTON_VARIENTS }
