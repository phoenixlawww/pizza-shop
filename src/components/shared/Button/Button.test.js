import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Button from './Button'

const testLabel = 'test label'
const testHandler = jest.fn()

test('renders correctly', () => {
  const { baseElement } = render(<Button label={testLabel} />)

  expect(baseElement).toMatchSnapshot()
})

test('handle click event correctly', () => {
  render(<Button label={testLabel} onClick={testHandler} />)

  const button = screen.getByText(testLabel)
  userEvent.click(button)

  expect(testHandler).toBeCalled()
})

test('handle disabled correctly', () => {
  render(<Button label={testLabel} onClick={testHandler} disabled />)

  const button = screen.getByText(testLabel)
  userEvent.click(button)

  expect(testHandler).not.toBeCalled()
})
