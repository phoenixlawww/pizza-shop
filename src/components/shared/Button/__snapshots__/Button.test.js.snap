// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <button
      class="s-button s-button-primary"
    >
      test label
    </button>
  </div>
</body>
`;
